adoc/chapters/programming_interface.adoc

```diff
@@ -5186,718 +5173,637 @@ If no template argument is provided, then the default allocator for the SYCL
+provided when allocating memory on the host.
+The range of the constructed SYCL [code]#buffer# is specified by the
+[code]#bufferRange# parameter provided.
 Zero or more properties can be provided to the constructed SYCL [code]#buffer#
 via an instance of [code]#property_list#.
 
-a@
-[source]
-----
-template <typename InputIterator>
-buffer(InputIterator first, InputIterator last,
-       const property_list& propList = {})
-----
-   a@ Create a new allocated 1D buffer initialized from the given elements
-      ranging from [code]#first# up to one before [code]#last#.
-      The data is copied to an intermediate memory position by the runtime.
-      Data is not written back to the same iterator set provided. However, if the [code]#buffer# has a valid non-const iterator specified via the member function [code]#set_final_data()#, data will be copied back to that iterator.
-      The constructed SYCL [code]#buffer# will use a default constructed [code]#AllocatorT# when allocating memory on the host.
-      Zero or more properties can be provided to the constructed SYCL [code]#buffer# via an instance of [code]#property_list#.
+_Effects (4):_ Equivalent to `buffer(hostData, bufferRange, AllocatorT{},
+propList)`.
 
-a@
-[source]
+'''
+
+.[apititle]#Construct from iterators#
+[source,role=synopsis,id=api:buffer-ctor-iterator]
 ----
-template <typename InputIterator>
+template <typename InputIterator>                                      (1)
+buffer(InputIterator first, InputIterator last, AllocatorT allocator,
+       const property_list& propList = {});
+
+template <typename InputIterator>                                      (2)
 buffer(InputIterator first, InputIterator last,
-       AllocatorT allocator = {},
-       const property_list& propList = {})
-----
-   a@ Create a new allocated 1D buffer initialized from the given elements
-      ranging from [code]#first# up to one before [code]#last#.
-      The data is copied to an intermediate memory position by the runtime.
-      Data is not written back to the same iterator set provided. However, if the [code]#buffer# has a valid non-const iterator specified via the member function [code]#set_final_data()#, data will be copied back to that iterator.
-      The constructed SYCL [code]#buffer# will use the [code]#allocator# parameter provided when allocating memory on the host.
-      Zero or more properties can be provided to the constructed SYCL [code]#buffer# via an instance of [code]#property_list#.
+       const property_list& propList = {});
+----
+_Effects (1):_ Create a new allocated 1D buffer initialized from the given
+elements ranging from [code]#first# up to one before [code]#last#.
+The data is copied to an intermediate memory position by the runtime.
+Data is not written back to the same iterator set provided.
+However, if the [code]#buffer# has a valid non-const iterator specified via the
+member function [code]#set_final_data()#, data will be copied back to that
+iterator.
```

Comment:
I don't think the part about the intermediate memory position is helpful.  We don't have this wording for the `hostMem` overload, and the inconsistency is weird.

I think we should re-use the same wording as the `hostMem` constructor, which I proposed changes to above:

```suggestion
_Remarks_: The buffer will not write back to any host memory
unless requested via the member function [code]#set_final_data()#.
```